source/jormungandr/jormungandr/scenarios/helper_classes/fallback_durations.py

```diff
@@ -103,9 +103,23 @@ def _get_duration(self, resp, place):
 
     @new_relic.distributedEvent("routing_matrix", "street_network")
     def _get_street_network_routing_matrix(self, origins, destinations):
-        return self._instance.get_street_network_routing_matrix(
-            origins, destinations, self._mode, self._max_duration_to_pt, self._request, **self._speed_switcher
-        )
+        try:
+            return self._instance.get_street_network_routing_matrix(
+                origins,
+                destinations,
+                self._mode,
+                self._max_duration_to_pt,
+                self._request,
+                **self._speed_switcher
+            )
+        except Exception as e:
+            logging.getLogger(__name__).error("Exception':{}".format(str(e)))
+            resp = response_pb2.StreetNetworkRoutingMatrix()
+            row = resp.rows.add()
+            routing = row.routing_response.add()
```

Comment:
It makes probably more sense that you set the error status at the level of resp... just for the sake of coherence (maybe we can use this field in the future to  raise the exception in the final response...)